File: 2021/Day7/Search in Rotated Sorted Array/Search_in_Rotated_Sorted_Array.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums)-1

        while low <= high:
            mid = (low+high)//2

            if nums[mid] == target:
                return mid

            if nums[mid] >= nums[low]:
                if target >= nums[low] and target < nums[mid]:
                    high = mid-1
                else:
                    low = mid+1

            else:
                if target > nums[mid] and target <= nums[high]:
                    low = mid+1
                else:
                    high = mid-1

        return -1

File: 2021/Day7/Search in Rotated Sorted Array/Search_in_Rotated_Sorted_Array.cpp
class Solution
{
public:
    int search(vector<int> &nums, int target)
    {
        int n = nums.size();
        int ans;
        bool found = false;
        for (int i = 0; i < n; i++)
        {
            if (nums[i] == target)
            {
                ans = i;
                found = true;
            }
        }
        if (found == false)
        {
            return -1;
        }
        return ans;
    }
};